feat: create versioned resources for windows setup

deepak1556 opened the pull request:
Fixes https://github.com/microsoft/vscode/issues/249239

deepak1556 committed:
feat: create versioned resources for windows setup

deepak1556 committed:
chore: use inno_updater to remove old installation

deepak1556 committed:
chore: remove old installation as part of setup

deepak1556 committed:
chore: update explorer-command

deepak1556 committed:
chore: prefer session-end

deepak1556 committed:
chore: uninst delete updating_version

deepak1556 committed:
chore: make session-ending write synchronous

deepak1556 committed:
chore: cleanup updateService.win32.ts

deepak1556 committed:
chore: invoke inno_updater gc path for non background update

deepak1556 committed:
chore: move session-end path to runtime

deepak1556 committed:
chore: use commit for updating_version

bpasero commented on src/vs/platform/native/electron-main/nativeHostMainService.ts:
```diff
@@ -748,6 +750,12 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 		return app.runningUnderARM64Translation;
 	}
 
+	@memoize
```
Not sure about this property in this service. The idea in this service is to provide methods to the renderer. And if its a static property it would better be suited in environment service.

deepak1556 commented on build/win32/code.iss:
```diff
@@ -1545,6 +1599,7 @@ begin
 
     if IsBackgroundUpdate() then
     begin
+      SaveStringToFile(ExpandConstant('{app}\updating_version'), {#Commit}, False);
```
```suggestion
      SaveStringToFile(ExpandConstant('{app}\updating_version'), '{#Commit}', False);
```

deepak1556 committed:
chore: fix invalid string

deepak1556 committed:
chore: set appUpdate path

deepak1556 committed:
chore: update inno_updater

deepak1556 committed:
chore: empty commit for testing

deepak1556 committed:
chore: some cleanups

1) Check for session-ending flag in appx and tunnel callsites
2) Move gc for background update to cleanup phase in updateservice
3) Set update state to ready when there is a running inno_setup

deepak1556 committed:
chore: disallow same version update

deepak1556 committed:
chore: disallow application launch in the middle of update

deepak1556 committed:
chore: empty commit for testing

deepak1556 committed:
chore: bump inno_updater

deepak1556 committed:
chore: empty commit for testing

deepak1556 committed:
chore: move gc to update startup

deepak1556 committed:
chore: move feature behind insider only check

deepak1556 committed:
chore: bump inno_updater

deepak1556 committed:
chore: bump explorer-command

deepak1556 committed:
fix: build

deepak1556 committed:
fix: gc for background update in system setup

deepak1556 committed:
chore: create separate cli entrypoints for build

deepak1556 committed:
fix: check for setup mutex created by inno

deepak1556 commented on src/vs/platform/update/electron-main/updateService.win32.ts:
```diff
@@ -99,6 +109,51 @@ export class Win32UpdateService extends AbstractUpdateService implements IRelaun
+				this.logService.error(`update#doCheckForUpdates - could not read ${updatingVersionPath}`, e);
+			} finally {
+				try {
+					await unlink(updatingVersionPath);
```
Let setup delete the file

deepak1556 committed:
chore: remove problematic updatingVersionPath deletion

deepak1556 committed:
chore: remove redundant update check

deepak1556 committed:
chore: bump inno_updater

deepak1556 committed:
chore: fix build

joaomoreno reviewed (dismissed):
Awesome!

deepak1556 committed:
chore: bump inno updater
